From 6f42c0fab7d5dbc092ae0d82d2fef3fa5af84244 Mon Sep 17 00:00:00 2001
From: Ricardo Cerqueira <ricardo@cyngn.com>
Date: Sun, 22 Nov 2015 21:16:20 -0500
Subject: [PATCH 1/2] libfm_jni: Skip loading FM firmware if requested

Not all boards require a firmware download for FM. Set
TARGET_QCOM_NO_FM_FIRMWARE if the device doesn't require
a fw image to run FM.

Adapted for libfm_jni by Sultan Qasim Khan.

Change-Id: I31fee137a59256c2b2907b5ff7212327299a9646
---
 libfm_jni/Android.mk            |  4 ++++
 libfm_jni/FmIoctlsInterface.cpp | 16 ++++++++++++++++
 2 files changed, 20 insertions(+)

diff --git a/libfm_jni/Android.mk b/libfm_jni/Android.mk
index f62e3f1..d398d08 100644
--- a/libfm_jni/Android.mk
+++ b/libfm_jni/Android.mk
@@ -23,6 +23,10 @@ LOCAL_SHARED_LIBRARIES := \
 #LOCAL_C_INCLUDES += $(TARGET_OUT_INTERMEDIATES)/KERNEL_OBJ/usr/include
 #LOCAL_ADDITIONAL_DEPENDENCIES := $(TARGET_OUT_INTERMEDIATES)/KERNEL_OBJ/usr
 
+ifeq ($(TARGET_QCOM_NO_FM_FIRMWARE),true)
+    LOCAL_CFLAGS += -DQCOM_NO_FM_FIRMWARE
+endif
+
 LOCAL_MODULE := libfmjni
 #include $(BUILD_SHARED_LIBRARY)
 
diff --git a/libfm_jni/FmIoctlsInterface.cpp b/libfm_jni/FmIoctlsInterface.cpp
index fbe035c..b577a02 100644
--- a/libfm_jni/FmIoctlsInterface.cpp
+++ b/libfm_jni/FmIoctlsInterface.cpp
@@ -69,6 +69,7 @@ int FmIoctlsInterface :: start_fm_patch_dl
             ALOGE("set FM_INIT_PROP done");
             if(ret != PROP_SET_SUCC)
                return FM_FAILURE;
+#ifndef QCOM_NO_FM_FIRMWARE
             ret = property_set(SCRIPT_START_PROP, SOC_PATCH_DL_SCRPT);
             if(ret != PROP_SET_SUCC)
                return FM_FAILURE;
@@ -81,6 +82,13 @@ int FmIoctlsInterface :: start_fm_patch_dl
                     usleep(INIT_WAIT_TIMEOUT);
                 }
             }
+#else
+            ret = property_set(FM_INIT_PROP, "1");
+            usleep(INIT_WAIT_TIMEOUT);
+            if(ret != PROP_SET_SUCC)
+               return FM_FAILURE;
+            init_success = 1;
+#endif
             if(!init_success) {
                 property_set(SCRIPT_STOP_PROP, SOC_PATCH_DL_SCRPT);
                 return FM_FAILURE;
@@ -100,12 +108,16 @@ int  FmIoctlsInterface :: close_fm_patch_dl
 {
     int ret;
 
+#ifndef QCOM_NO_FM_FIRMWARE
     ret = property_set(SCRIPT_STOP_PROP, SOC_PATCH_DL_SCRPT);
     if(ret != PROP_SET_SUCC) {
         return FM_FAILURE;
     }else {
         return FM_SUCCESS;
     }
+#else
+    return FM_SUCCESS;
+#endif
 }
 
 int  FmIoctlsInterface :: get_cur_freq
@@ -176,6 +188,7 @@ int  FmIoctlsInterface :: set_calibration
     struct v4l2_ext_controls v4l2_ctls;
     char cal_data[CAL_DATA_SIZE] = {0};
 
+#ifndef QCOM_NO_FM_FIRMWARE
     cal_fp = fopen(CALIB_DATA_NAME, "r");
     if(cal_fp != NULL) {
        if(fread(&cal_data[0], 1, CAL_DATA_SIZE, cal_fp)
@@ -198,6 +211,9 @@ int  FmIoctlsInterface :: set_calibration
     }else {
         return FM_FAILURE;
     }
+#else
+    return FM_SUCCESS;
+#endif
 }
 
 int  FmIoctlsInterface :: get_control
-- 
2.9.3

